decidim-core/app/commands/decidim/gallery_methods.rb

```diff
@@ -36,17 +32,31 @@ def gallery_invalid?
       false
     end
 
-    def create_gallery
+    def create_gallery(first_weight: 0)
+      weight = first_weight
```

Comment:
Excellent! Today, when we were looking to this PR, one "wish" that was mentioned is about being able to reorder galleries/multiple attachments... Shouldn't be that difficult if we have the attribute here already as far as I understand 😄

Reply:
> But the current implementation for the proposal card image is quite clunky and hack(ish)

Yes, I agree: we don't mention it to the user, so it's a weird behavior. Also, as you mention, it means that for changing it the user need to delete the rest first.

For now, this PR is a great improvement for a good starting point to solve these kinds of issues.